Unix/agent/agent.c

```diff
@@ -85,6 +88,56 @@ static void FUNCTION_NEVER_RETURNS err(const ZChar* fmt, ...)
     exit(1);
 }
 
+void ResetLog()
+{    
+    char path[PAL_MAX_PATH_SIZE];
+    Conf* conf;
+
+    /* Form the configuration file path */
+    Strlcpy(path, OMI_GetPath(ID_CONFIGFILE), sizeof(path));
+
+    /* Open the configuration file */
+    conf = Conf_Open(path);
+    if (!conf)
+    {
+        err(ZT("failed to open configuration file: %s"), scs(path));
+    }
+
+    /* For each key=value pair in configuration file */
+    for (;;)
+    {
+        const char* key;
+        const char* value;
+        int r = Conf_Read(conf, &key, &value);
+
+        if (r == -1)
```

Comment:
can r value be always -1? in that case we will go in infinite loop

Reply:
"r" value cannot be always -1, it will only in error scenarios. In that case we breaking the loop.